ntbs-service/Services/SpecimenService.cs

```diff
@@ -111,6 +114,29 @@ public async Task<IEnumerable<SpecimenMatchPairing>> GetAllSpecimenPotentialMatc
             }
         }
 
+        public async Task<IEnumerable<(string LegacyId, string ReferenceLaboratoryNumber)>>
+            GetLegacyReferenceLaboratoryMatches(IEnumerable<string> legacyIds)
+        {
+            // The table we're referencing here has legacyIds stored as INTs (since they are all ETS ids)
+            // Therefore we need to convert to and from strings
+            var intIds = legacyIds
+                .Select(id => int.TryParse(id, out var intId) ? intId : (int?)null)
+                .Where(id => id.HasValue)
+                .Select((id => id.Value));
+            using (var connection = new SqlConnection(_specimenMatchingDbConnectionString))
+            {
+                connection.Open();
+                return (await connection.QueryAsync<(int LegacyId, string ReferenceLaboratoryNumber)>(
+                        SpecimenQueryHelper.LegacyReferenceLaboratoryMatchesQuery,
+                        new { Ids = intIds }))
+                    .Select(tuple =>
+                    {
+                        var legacyId = tuple.LegacyId.ToString();
+                        return (LegacyId: legacyId, tuple.ReferenceLaboratoryNumber);
+                    });

```

Comment:
Likewise, I think this would be more readable if it was split into two steps, i.e.
```
var queryResults = await connection.QueryAsync(...);
return queryResults.Select(...);
```

Reply:
Yep, done!